apps/arena/lib/arena/game_updater.ex

```diff
@@ -1873,6 +1911,52 @@ defmodule Arena.GameUpdater do
     end
   end
 
+  defp add_players_to_respawn_queue(game_state, %{game: %{game_mode: :DEATHMATCH}} = game_config) do
+    death_players =
+      game_state.players
+      |> Enum.filter(fn {_player_id, player} ->
+        not Player.alive?(player)
```

Comment:
couldn't we just check here if the player is not in the respawn queue already? That way we avoid the 'if' below and another iteration over the dead players.

Reply:
We can just remove the first iteration and move everything to the second iteration